llm_memory/components/embedding_provider.py

```
@@ -927,19 +928,33 @@ async def _get_embeddings_with_batch(
+                    batches = [texts[i : i + batch_size] for i in range(0, len(texts), batch_size)]
+
+                    async def run_batch(batch):
+                        async with sem:
+                            for attempt in range(4):
+                                try:
+                                    return await provider.get_embeddings(batch)
+                                except Exception as e:
+                                    if self._is_rate_limit_error(e) and attempt < 3:
+                                        delay = (2**attempt) + (attempt * 2)
+                                        self.logger.warning(
+                                            f"批量向量化遇到429限流，等待{delay}秒后重试 "
```

Ruff (0.16.2)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)